modes/mpdel/evil-collection-mpdel.el

```diff
@@ -0,0 +1,94 @@
+;;; evil-collection-mpdel.el --- Evil bindings for mpdel -*- lexical-binding: t -*-
+
+;; Copyright (C) 2021 Ruoyu Feng
+
+;; Author: Ruoyu Feng <mail@vonfry.name>
+;; Maintainer: James Nguyen <james@jojojames.com>
+;; Pierre Neidhardt <mail@ambrevar.xyz>
+;; URL: https://github.com/emacs-evil/evil-collection
+;; Version: 0.0.4
+;; Package-Requires: ((emacs "25.1"))
+;; Keywords: evil, tools, mpd
+
+;; This file is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published
+;; by the Free Software Foundation; either version 3, or (at your
+;; option) any later version.
+;;
+;; This file is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; For a full copy of the GNU General Public License
+;; see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Evil bindings for mpdel.
+
+;;; Code:
+
+(require 'mpdel nil t)
+(require 'evil-collection)
+
+(defvar mpdel-core-map)
+
+(defconst evil-collection-mpdel-maps '(mpdel-browser-mode mpdel-playlist-mode
+                                       mpdel-song-mode mpdel-browser-mode
+                                       mpdel-playlist-current-playlist-mode
+                                       mpdel-playlist-stored-playlist-mode))
+
+;;;###autoload
+(defun evil-collection-mpdel-setup ()
+  "Set up `evil' bindings for `mpdel'."
+
+  (evil-collection-set-readonly-bindings 'mpdel-core-map)
+  (dolist (mode evil-collection-mpdel-maps)
+    (evil-set-initial-state mode 'normal))
+  (evil-collection-define-key 'normal 'mpdel-core-map
+    "p"  #'libmpdel-playback-play-pause
+    "{"  #'libmpdel-playback-next
+    "}"  #'libmpdel-playback-previous
+    "a"  #'mpdel-core-add-to-current-playlist
+    "A"  #'mpdel-core-add-to-stored-playlist
+    "r"  #'mpdel-core-replace-current-playlist
+    "R"  #'mpdel-core-replace-stored-playlist
+    "i"  #'mpdel-core-insert-current-playlist
+    "gD" #'mpdel-core-dired
+    "gA" #'mpdel-core-open-artists
+    "gL" #'mpdel-core-open-stored-playlists
+    "ss" #'mpdel-core-search-by-title
+    "sl" #'mpdel-core-search-by-album
+    "sr" #'mpdel-core-search-by-artist
+    "gp" #'navigel-open-parent
+    "j"  #'next-line
+    "k"  #'previous-line
+    "+"  #'mpdel-core-volume-increase
+    "-"  #'mpdel-core-volume-decrease
+    "C"  #'libmpdel-connect-profile
+
+    "gb" #'mpdel-browser-open
+
+    "go" #'mpdel-song-open
+    "gr" #'mpdel-song-refresh
+    "I"  #'mpdel-song-play
```

Comment:
This doesn't seem super intuitive but I've never used this package so maybe it's OK.

Consider maybe adding a wrapper function for toggle play? Maybe that'd be more appropriate for the original package.

Reply:
> Consider maybe adding a wrapper function for toggle play?

The key `p` is used for that. The three functions are not used to toggle play. `song-open` is used to open a buffer to show current playing song information. `song-refresh` is used to refresh buffer content. `song-play` is used to play a single song under cursor in browser.